apps/challenges/aws_utils.py

```diff
@@ -1294,6 +1294,341 @@ def delete_log_group(log_group_name):
+            return
+
+    try:
+        client.delete_role(
+            RoleName=node_group_role_name,
+        )
+    except ClientError as e:
+        logger.exception(e)
+        return
+
+    eks_role_name = "evalai-code-upload-eks-role-{}".format(environment_suffix)
+    try:
+        # Detach AWS managed EKS cluster policy to the role
+        client.detach_role_policy(
+            RoleName=eks_role_name,
+            PolicyArn=settings.EKS_CLUSTER_POLICY,
+        )
+    except ClientError as e:
+        logger.exception(e)
+        return
+
+    try:
+        client.delete_role(
+            RoleName=eks_role_name,
+        )
+    except ClientError as e:
+        logger.exception(e)
+        return
+
+    try:
+        serializer = ChallengeEvaluationClusterSerializer(
+            challenge_evaluation_cluster,
+            data={
+                "active": False,
+            },
+            partial=True,
+        )
+        if serializer.is_valid():
+            serializer.save()
+            return serializer.data
+    except Exception as e:
+        logger.exception(e)
+        return
+
+
+def delete_evaluation_clusters(queryset):
+    """
+    The function called by the admin action method to delete all the selected evaluation clusters.
+
+    Calls the delete_setup_eks_cluster method. Before calling, verifies that the evaluation cluster is active.
+
+    Parameters:
+    queryset (<class 'django.db.models.query.QuerySet'>): The queryset of selected challenges in the django admin page.
+
+    Returns:
+    dict: keys-> 'count': the number of evaluation clusters successfully deleted.
+                 'failures': a dict of all the failures with their error messages and the challenge pk
+    """
+    from .models import ChallengeEvaluationCluster
+
+    if settings.DEBUG:
+        failures = []
+        for challenge in queryset:
+            failures.append(
+                {
+                    "message": "Workers cannot be deleted on AWS ECS service in development environment",
+                    "challenge_pk": challenge.pk,
+                }
+            )
+        return {"count": 0, "failures": failures}
+
+    count = 0
+    failures = []
+    for challenge in queryset:
+        challenge_evaluation_cluster = ChallengeEvaluationCluster.objects.get(
+            challenge=challenge
+        )
+        if not challenge.is_docker_based:
+            response = "Sorry. This feature is not available for non code upload/docker based challenges."
+            failures.append(
+                {"message": response, "challenge_pk": challenge.pk}
+            )
+        elif challenge_evaluation_cluster.active:
+            serialized_obj = serializers.serialize("json", [challenge])
+            response = delete_setup_eks_cluster(serialized_obj)
+            if not response:
+                failures.append(
+                    {
+                        "message": "Error deleting Evaluation Cluster",
```

Comment:
This error message is very generic. There are a bunch of calls in this workflow, and better to be more specific with where the failure happened, and show that to the user.